tls/ttls/net.c

```diff
@@ -449,36 +142,37 @@ void mbedtls_net_usleep( unsigned long usec )
  */
 int mbedtls_net_recv( void *ctx, unsigned char *buf, size_t len )
 {
-    int ret;
-    int fd = ((mbedtls_net_context *) ctx)->fd;
-
-    if( fd < 0 )
-        return( MBEDTLS_ERR_NET_INVALID_CONTEXT );
-
-    ret = (int) read( fd, buf, len );
+    struct socket *sock = ((mbedtls_net_context *)ctx)->socket;
+    struct msghdr msg;
+    struct iovec iov;
+    mm_segment_t oldfs;
+    int size;
+
+    if(sock == NULL) {
+        return -1;
+    }
```

Comment:
From CodingStyle:

Do not unnecessarily use braces where a single statement will do.              
        
        if (condition)
                action();